packages/back-end/src/routers/demo-datasource-project/demo-datasource-project.controller.ts

```diff
@@ -166,11 +166,15 @@ export const postDemoDatasourceProject = async (
     EventAuditUserForResponseLocals
   >
 ) => {
+  const context = getContextFromReq(req);
+
   req.checkPermissions("manageProjects", "");
   req.checkPermissions("createDatasources", "");
-  req.checkPermissions("createMetrics", "");
   req.checkPermissions("createAnalyses", "");
-  const context = getContextFromReq(req);
+  if (!context.permissions.canCreateMetric({})) {
```

Comment:
We have a `demoProjId` below that we should use here since that's where the metric will be created.